pycontrails/core/aircraft_performance.py

```diff
@@ -476,3 +490,62 @@ class AircraftPerformanceData:
     thrust: npt.NDArray[np.float_]
     engine_efficiency: npt.NDArray[np.float_]
     rocd: npt.NDArray[np.float_]
+
+
+# --------------------------------
+# Grid aircraft performance models
+# --------------------------------
+
+
+@dataclasses.dataclass
+class AircraftPerformanceGridParams(ModelParams):
+    """Parameters for :class:`AircraftPerformanceGrid`."""
+
+    #: Fuel type
+    fuel: fuel.Fuel = dataclasses.field(default_factory=fuel.JetA)
+
+    #: ICAO code designating simulated aircraft type.
+    #: Can be overridden by including ``aircraft_type`` attribute in source data
+    aircraft_type: str = "B737"
+
+    #: Mach number, [:math:`Ma`]
+    #: If ``None``, a nominal cruise value is determined by the implementation.
+    #: Can be overridden by including a ``mach_number`` key in source data
+    mach_number: float | None = None
+
+    #: Aircraft mass, [:math:`kg`]
+    #: If ``None``, a nominal value is determined by the implementation.
+    #: Can be overridden by including an ``aircraft_mass`` key in source data
+    aircraft_mass: float | None = None
+
+
+class AircraftPerformanceGrid(Model):
+    """
+    Support for standardizing aircraft performance methodologies on a grid.
+
+    Currently just a container until additional models are implemented.
+    """
+
+    @overload
+    @abc.abstractmethod
+    def eval(self, source: GeoVectorDataset, **params: Any) -> GeoVectorDataset:
+        ...
+
+    @overload
+    @abc.abstractmethod
+    def eval(self, source: MetDataset | None = ..., **params: Any) -> MetDataset:
+        ...
+
+    @abc.abstractmethod
+    def eval(
+        self, source: GeoVectorDataset | MetDataset | None = None, **params: Any
+    ) -> GeoVectorDataset | MetDataset:
+        """Evaluate the aircraft performance model."""
+
+
+@dataclasses.dataclass
+class AircraftPerformanceGridData(Generic[ArrayOrFloat]):
+    """Store the computed aircraft performance metrics for nominal cruise conditions."""
+
+    fuel_flow: ArrayOrFloat
```

Comment:
Should we add units here for sake of documentation?

Reply:
ccd293abe4bb4d50c063843b46e3b41eebebe273